src/main/java/org/jenkinsci/plugins/preSCMbuildstep/PreSCMBuildStepsWrapper.java

```diff
@@ -55,15 +55,18 @@ public class PreSCMBuildStepsWrapper extends BuildWrapper {
 
     public final boolean failOnError;
 
+    public final String actionOnError;
+
     /**
      * Constructor taking a list of buildsteps to use.
      *
      * @param buildstep list of but steps configured in the UI
      */
     @DataBoundConstructor
-    public PreSCMBuildStepsWrapper(ArrayList<BuildStep> buildstep, boolean failOnError) {
+    public PreSCMBuildStepsWrapper(ArrayList<BuildStep> buildstep, boolean failOnError, String actionOnError) {
```

Comment:
Need to add a `DataBoundSetter` rather than extending the `DataBoundConstructor`.  That retains compatibility with any callers that might have existed for the original constructor and is the preferred pattern in plugins.

Also need to merge with the latest changes from the master branch and resolve the spotbugs warning that `failOnError` is no longer read.  I suspect that spotbugs warning is indicating a real problem, since the previous user configuration of `failOnError` should be honored on upgrade to a plugin version that is using this enhancement.